librz/core/cmd/cmd_print.c

```diff
@@ -6570,3 +6474,82 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
 RZ_IPI RzCmdStatus rz_print_value8_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
 	return print_value_size(core, state, argc, argv, 8);
 }
+
+static void printraw(RzCore *core, int len) {
+	int obsz = core->blocksize;
+	int restore_obsz = 0;
+	if (len != obsz) {
+		if (!rz_core_block_size(core, len)) {
+			len = core->blocksize;
+		} else {
+			restore_obsz = 1;
+		}
+	}
+	rz_print_raw(core->print, core->offset, core->block, len);
+	if (restore_obsz) {
+		(void)rz_core_block_size(core, obsz);
+	}
+	core->cons->newline = core->cmd_in_backticks ? false : true;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_raw_handler(RzCore *core, int argc, const char **argv) {
+	int len = argc > 1 ? rz_num_math(NULL, argv[1]) : strlen((const char *)core->block);
+	if (len < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	printraw(core, len);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_raw_colors_handler(RzCore *core, int argc, const char **argv) {
+	int len = argc > 1 ? rz_num_math(NULL, argv[1]) : strlen((const char *)core->block);
+	if (len < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	colordump(core, core->block, len);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_raw_gunzip_handler(RzCore *core, int argc, const char **argv) {
+	int outlen = 0;
+	ut8 *out;
+	out = rz_inflate(core->block, core->blocksize, NULL, &outlen);
+	if (!out) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_memcat((const char *)out, outlen);
+	free(out);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_raw_gunzip_verbose_handler(RzCore *core, int argc, const char **argv) {
+	int outlen = 0;
+	int inConsumed = 0;
+	ut8 *out;
+	out = rz_inflate(core->block, core->blocksize, &inConsumed, &outlen);
+	if (!out) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_printf("consumed: %d produced: %d\n", inConsumed, outlen);
+	rz_cons_memcat((const char *)out, outlen);
+	free(out);
+	return RZ_CMD_STATUS_OK;
+}
```

Comment:
Put this in a separate function. And call it both here and in the previous function. They are the same except a verbose print.